include/battle_controllers.h

```diff
@@ -64,11 +64,56 @@ enum {
     REQUEST_TOUGH_RIBBON_BATTLE,
 };
 
+// Accessors for gBattleControllerExecFlags.
+//
+// These are provided for documentation purposes, to make the battle 
+// controller internals and the link communication internals more 
+// legible. Several of these have functions that you should call 
+// (e.g. MarkBattlerForControllerExec) instead of using these macros 
+// directly.
+
+#define MARK_BATTLE_CONTROLLER_ACTIVE_ON_LOCAL(battlerId) \
+   gBattleControllerExecFlags |= gBitTable[battlerId]
+
+#define MARK_BATTLE_CONTROLLER_IDLE_ON_LOCAL(battlerId) \
+   gBattleControllerExecFlags &= ~gBitTable(battlerId)
+
+#define IS_BATTLE_CONTROLLER_ACTIVE_ON_LOCAL(battlerId) \
+   (gBattleControllerExecFlags & gBitTable[battlerId])
+
+#define MARK_BATTLE_CONTROLLER_MESSAGE_OUTBOUND_OVER_LINK(battlerId) \
+   gBattleControllerExecFlags |= gBitTable[battlerId] << (32 - MAX_BATTLERS_COUNT)
+
+#define MARK_BATTLE_CONTROLLER_MESSAGE_SYNCHRONIZED_OVER_LINK(battlerId) \
+   gBattleControllerExecFlags &= ~((1 << 28) << (battlerId))
+
+#define MARK_BATTLE_CONTROLLER_ACTIVE_FOR_PLAYER(battlerId, playerId) \
+   gBattleControllerExecFlags |= gBitTable[battlerId] << ((playerId) << 2)
+
+#define MARK_BATTLE_CONTROLLER_IDLE_FOR_PLAYER(battlerId, playerId) \
+   gBattleControllerExecFlags &= ~(gBitTable[battlerId] << ((playerId) * 4))
+
+#define IS_BATTLE_CONTROLLER_ACTIVE_FOR_PLAYER(battlerId, playerId) \
+   (gBattleControllerExecFlags & (gBitTable[battlerId] << ((playerId) * 4)))
+
+// This actually checks if a specific controller is active on any player or if 
+// *any* controller is pending sync over link communications, but the macro name 
+// can only be so specific before it just gets ridiculous.
+#define IS_BATTLE_CONTROLLER_ACTIVE_OR_PENDING_SYNC_ANYWHERE(battlerId) \
+   (gBattleControllerExecFlags & ( \
+      (gBitTable[battlerId])       \
+    | (0xF << 28)                  \
+    | (gBitTable[battlerId] << 4)  \
+    | (gBitTable[battlerId] << 8)  \
+    | (gBitTable[battlerId] << 12) \
+   ))
+
 // Special arguments for Battle Controller functions.
 
-enum { // Values given to the emit functions to choose gBattleBufferA or gBattleBufferB
-    BUFFER_A,
-    BUFFER_B
+enum {
+   BATTLELINKCOMMTYPE_ENGINE_TO_CONTROLLER, // gBattleBufferA
+   BATTLELINKCOMMTYPE_CONTROLLER_TO_ENGINE, // gBattleBufferB
+   BATTLELINKCOMMTYPE_CONTROLLER_BECOMING_IDLE
 };
```

Comment:
These new names are a bit of a mouthful, esp. for how often they're used. At a minimum, we have other names that shorten `BATTLE` as a prefix to `B_`, and `LINKCOMM` could be shortened to `COMM` (which also avoids implying it's related to multiplayer battles. That's often what "link" is associated with)